config/config_test.go

```diff
@@ -89,34 +89,40 @@ var expectedConf = &Config{DefaultedConfig{
 			Scheme:      "https",
 
 			DNSSDConfigs: []*DNSSDConfig{
-				{DefaultedDNSSDConfig{
+				{
 					Names: []string{
 						"first.dns.address.domain.com",
 						"second.dns.address.domain.com",
 					},
 					RefreshInterval: Duration(15 * time.Second),
-				}},
-				{DefaultedDNSSDConfig{
+				},
+				{
 					Names: []string{
 						"first.dns.address.domain.com",
 					},
 					RefreshInterval: Duration(30 * time.Second),
-				}},
+				},
 			},
 
 			RelabelConfigs: []*RelabelConfig{
-				{DefaultedRelabelConfig{
+				{
 					SourceLabels: clientmodel.LabelNames{"job"},
 					Regex:        &Regexp{*regexp.MustCompile("(.*)some-[regex]$")},
 					Separator:    ";",
 					Action:       RelabelDrop,
-				}},
+				},
 			},
-		}},
+		},
 	},
-}, ""}
+	original: "",
+}
 
 func TestLoadConfig(t *testing.T) {
+	// Parse a valid file that sets a global scrape timeout.
+	if _, err := LoadFromFile("testdata/glob_timeout.good.yml"); err != nil {
```

Comment:
With the name of the file, the reader will expect a glob. But you mean "global", don't you?

Is this to test that the default conf is not overwritten by it? In that case, mention that in a comment.